### PR TITLE
Update supported versions matrix

### LINKED ISSUE #134: Bug: Cannot create dbt profile from Airflow connection
Version: 2.0.1
Error: 
```log
File "/usr/local/airflow/.local/lib/python3.11/site-packages/airflow_dbt_python/utils/configs.py", line 529, in create_dbt_profile
    profile = Profile.from_raw_profile_info(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Profile.from_raw_profile_info() got an unexpected keyword argument 'user_config'
```

### LINKED ISSUE #135: No module found dbt.semver
Hi

I tried running with dbt-core 1.8 yesterday on MWAA and ran into No module name 'dbt.semver'... has any integration testing been done yet for dbt-core 1.8?.. Do you know when the release will happen?

### PR DESCRIPTION
**Pull request by millin**
fixes #134
fixes #135

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -22,20 +22,18 @@ jobs:
...
           - '3.9'
           - '3.8'
         airflow-version:
-          - '2.7.2'
-          - '2.6.3'
-          - '2.5.3'
-          - '2.4.3'
+          - '2.9.2'
+          - '2.8.4'
+          - '2.7.3'
```
Comment (tomasfarias) [inline .github/workflows/ci.yaml:33]: We'll probably want a policy for Airflow version support.

For Python versions it's easy: We aim to support whatever dbt supports, which [currently](https://github.com/dbt-labs/dbt-core/blob/main/core/setup.py#L5) has only a >=3.8. restriction.

For Airflow, I'd say we definitely want to test against [latest official minor version](https://docs.aws.amazon.com/mwaa/latest/userguide/airflow-versions.html#airflow-versions-official) (2.9.x), and ideally also [latest MWAA-supported version](https://docs.aws.amazon.com/mwaa/latest/userguide/airflow-versions.html#airflow-versions-official) (2.8.1). Astronomer pretty much supports new Airflow version as they come out, so we are covered there already.


Any other hosted service we should consider prioritizing? If not, I'd consider just supporting 2.9.2 and 2.8.1 here. There is a tradeoff between as much support as possible and CI times and dependency resolution. So, I'd try to keep the number of "tested" supported versions as small as possible.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -22,20 +22,18 @@ jobs:
...
         dbt-version:
+          - 1.8
           - 1.7
```
Comment (tomasfarias) [inline .github/workflows/ci.yaml:36]: 2 latest dbt versions is a fair policy :+1:

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -22,20 +22,18 @@ jobs:
...
-          - 1.6
-          - 1.5
-          - 1.4
         exclude:
           # Incompatible combinations
           - python-version: 3.11
```
Comment (tomasfarias) [inline .github/workflows/ci.yaml:39]: If we go ahead with the changes above, should be fine to remove these two excludes.

**Comment by tomasfarias** [comment]
Tried testing locally, but it seems like some new flags need to be supported starting with dbt 1.7, and `user_config` parameter has been removed from `Profile`. I managed to get some tests to pass by adding the following flags in `utils/config.py`:

```python
        flags_dict["PARTIAL_PARSE_FILE_DIFF"] = self.partial_parse_file_diff
        flags_dict["USE_EXPERIMENTAL_PARSER"] = self.use_experimental_parser
        flags_dict["CACHE_SELECTED_ONLY"] = self.cache_selected_only
        flags_dict["PARTIAL_PARSE_FILE_PATH"] = self.partial_parse_file_path
```
(And also the corresponding class attributes)

As well as dropping `user_config=user_config` when calling `from_raw_profile_info`.

Dag tests (`tests/dags`) are passing, but other tests seem to fail to select any models to run. Something seems to have changed with latter versions of dbt but I haven't figured out what :thinking:

**Comment by millin** [comment]
Sorry, I created this pr prematurely. 
I've already fixed most of the bugs locally, but haven't pushed yet. I'll try to finish on Monday.

**Comment by tomasfarias** [comment]
> Sorry, I created this pr prematurely.
I've already fixed most of the bugs locally, but haven't pushed yet. I'll try to finish on Monday.

No problem! Happy to review next week and get this released.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (millin) [reply .github/workflows/ci.yaml:33]: I would like to support Airflow 2.7.x as it is still in use by [Cloud Composer](https://cloud.google.com/composer/docs/concepts/overview)

File: README.md [hunk 4]
```diff
@@ -66,6 +66,12 @@ Add *airflow-dbt-python* to your `requirements.txt` file and edit your Airflow e
 
 Read the [documentation](https://airflow-dbt-python.readthedocs.io/en/latest/getting_started.html#installing-in-mwaa) for more a more detailed AWS MWAA installation breakdown.
 
+## In GCP Cloud Composer
```
Comment (tomasfarias) [inline README.md:69]: Thanks! Great updates!

**Review by tomasfarias — APPROVED** [review]

**Comment by tomasfarias** [comment]
Will release shortly. Thanks for the great work @millin

**Pull request merged**